pkg/reconciler/instances/serverless/nodeport.go

```diff
@@ -0,0 +1,155 @@
+package serverless
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"math/rand"
+)
+
+var _ service.Action = ResolveDockerRegistryNodePort{}
+
+const (
+	dockerRegistryNodePort = 32_137
+
+	//Available ports according to documentation https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport
+	maxNodePort = 32_767
+	minNodePort = 30_000
+)
+
+const (
+	dockerRegistryService      = "serverless-docker-registry"
+	dockerRegistryNodePortPath = "global.registryNodePort"
+	dockerRegistryPortName     = "http-registry"
+)
+
+type ResolveDockerRegistryNodePort struct {
+}
+
+func (n ResolveDockerRegistryNodePort) Run(svcCtx *service.ActionContext) error {
+	k8sClient, err := svcCtx.KubeClient.Clientset()
+	if err != nil {
+		return errors.Wrap(err, "while getting clientset")
+	}
+	svc, err := getService(svcCtx.Context, k8sClient, svcCtx.Task.Namespace, dockerRegistryService)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("while checking if %s service is installed on cluster", dockerRegistryService))
+	}
+
+	if svc != nil {
+		if isDefaultNodePortValue(svc) {
+			return nil
+		}
+		currentNodePort := getNodePort(svc)
+		setNodePortOverride(svcCtx.Task.Configuration, dockerRegistryNodePortPath, currentNodePort)
```

Comment:
shouldn't the new nodeport value be drawn in this case? I tested the code and it keeps applying the same nodeport ```
kubeClient failed to update Service 'serverless-docker-registry' (namespace: kyma-system)  with strategy 'PATCH': cannot patch "serverless-docker-registry" with kind Service: Service "serverless-docker-registry" is invalid: spec.ports[0].nodePort: Invalid value: 32137: provided port is already allocated
```